draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -654,476 +526,437 @@
+	will not be started, because uCDN cannot control the timing of that activity. 
+	Processing could, for example, start after the POST is sent by uCDN but before that request is processed by dCDN.
    </t>
 
-   </section>
-   
-   <section title="Checking Status" anchor="checking-status"><t>
-   uCDN has two ways to check the progress of CI/T Commands it has
-   issued to dCDN, as described in Sections <xref target="polling-collections"/> and 
-   <xref target="polling-trigger-resources"/>.</t>
+   <t>
+   	If cancellation of an "active" or "processed" trigger is accepted by dCDN, dCDN SHOULD stop processing the trigger.
+	However, as with the cancellation of a "pending" trigger, dCDN does not guarantee that the trigger processing 
+	doesn't run to completion in the meantime.
+   </t>
 
    <t>
-   To allow uCDN to check for changes in the status of a Trigger
-   Status Resource or collection of Trigger Status Resources without
-   refetching the whole resource or collection, dCDN SHOULD include
-   entity-tags (ETags) for uCDN to use as cache validators, as
-   defined in <xref target="RFC9110"/>.</t>
+	If uCDN cannot stop the trigger processing immediately after receiving the request from uCDN to do so, it MUST set the trigger  
+	state to "cancelling" and provide this state in the trigger representation in its response. 
+	If the trigger processing is stopped before its normal completion, the trigger state MUST be set to "cancelled".
+   </t>
 
    <t>
-   dCDN SHOULD use the cache control headers for responses to GETs
-   for Trigger Status Resources and Collections to indicate the
-   frequency at which it recommends that uCDN should poll for
-   change.</t>
+   	Cancellation of a "complete", "failed" or "cancelled" trigger requires no processing in dCDN. Its state MUST NOT be changed.
+   </t>
 
-   <section title="Polling Trigger Status Resource Collections" anchor="polling-collections">
+   </section>
+   
+   <section title="Checking Status" anchor="checking-status">
    <t>
-   uCDN can fetch the collection of its Trigger Status Resources or
-   filtered views of that collection.
+   	uCDN has two ways to check the progress of its triggers' processing, as described in <xref target="polling-collections"/> and 
+	<xref target="polling-trigger-resources"/>.
    </t>
 
    <t>
-   This makes it possible to poll the status of all CI/T Trigger
-   Commands in a single request. If dCDN moves a Trigger Status
-   Resource from the active to the completed collection, uCDN can
-   fetch the result of that activity.</t>
+	To allow uCDN to use client-side caching of all triggers and trigger collections,  dCDN SHOULD include an ETag response header 
+	with every resource representation it sends. dCDN SHOULD also respond with the HTTP code 304 ("Not Modified") without a response 
+	body in response to conditional resource requests that use "If-None-Match" and/ir "If-Not-Modified-Since" request headers 
```

Comment:
and/ir => and/or

Reply:
There is no "If-Not-Modified-Since" header.  There is "If-Unmodified-Since", which is less commonly used than "If-Modified-Since".

I'm not sure this is a good idea.  IMS / IUS headers assume you have a valid Last-Modified date, and we haven't discussed that here.  We are prescribing ETags as the method of doing revalidation, so this means "If-None-Match", and the less commonly used "If-Match".

Reply:
changed it to If-Modified-Since
The origin text allowed for ETags only, why wouldn't we allow for IMS as well? 
It makes sense to me to use it in the context of periodic polling

Reply:
Proper support of IMS requires a Last-Modified response header, just like proper support of INM requires an ETag response header.  We have specified the use of ETag but not Last-Modified in the document.  So, at a minimum, the document would need to be amended to mandate the use of either ETag or Last-Modified, and not just ETag.

The only concern I might have is if we're placing too many implementation barriers around what the uCDN needs to support.  So if either is possible, then uCDN would need to be developed to accept both, and conditionally send either IMS or INM, depending on what the dCDN decided to implement.  Either that, or the dCDN will now need to support both ETag and LM to give clients flexibility.

Reply:
Amended to ask dCDN to support at least on of the two methods: 

To allow uCDN to use client-side caching of all triggers and trigger collections,  each resource representation snet by dCDN SHOULD include at least one of the following response headers: "ETag" or "Last-Modified".  
 The dCDN should respond with the HTTP 304 ("Not Modified") status code and no response body for conditional resource requests using the 'If-None-Match' and/or 'If-Modified-Since' headers, as specified in Section 13 of <xref target="RFC9110"/>, if it does not have a more recent resource representation.

Reply:
I still have some concerns for some additional revisions that appear necessary, but I have carried them over to the new PR.